core/src/main/scala/cats/syntax/alternative.scala

```diff
@@ -49,6 +49,30 @@ final class SeparateOps[F[_], G[_, _], A, B](private val fgab: F[G[A, B]]) exten
                F: Monad[F],
                A: Alternative[F],
                G: Bifoldable[G]): (F[A], F[B]) = A.separate[G, A, B](fgab)
+
+  /**
+   * Separate the inner foldable values into the "lefts" and "rights".
+   * A variant of [[separate]] that is specialized
+   * for Fs that have Foldable instances
+   * which allows for a single-pass implementation
+   * (as opposed to {{{separate}}} which is 2-pass).
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> val l: List[Either[String, Int]] = List(Right(1), Left("error"))
+   * scala> l.separateFoldable
+   * res0: (List[String], List[Int]) = (List(error),List(1))
+   * }}}
+   */
+  def separateFoldable(implicit AA: Alternative[F], G: Bifoldable[G], FF: Foldable[F]): (F[A], F[B]) =
```

Comment:
It really bothers me now that these instance parameters are in a different order from the ones for `separate`, and that they don't follow the standard practice of matching the order the type parameters appear in, but I think matching the mainline (which is also wrong) is more important than internal consistency.

Reply:
so you mean the order should have been like this?

`implicit A: Alternative[F], F: Foldable[F], G: Bifoldable[G]`

Reply:
Oh, sorry that wasn't very clear. I meant

```scala
FF: Foldable[F],  A: Alternative[F], G: Bifoldable[G]
```
…which is the same as the syntax method on mainline (which is correct), but not the type class method (which is wrong, but unfixable for now). We only have the syntax method in this PR, so we should use `Foldable, Alternative, Bifoldable` both because it's consistent with mainline and it matches our general convention.

Reply:
I see, thank you.